opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/EventConfRestServiceIT.java

```diff
@@ -614,4 +618,103 @@ public void testUpdateEventConfEvent() throws Exception {
+
+        InputStream is = getClass().getResourceAsStream(path);
+        assertNotNull("Resource not found: " + path, is);
+
+        Attachment attachment = mock(Attachment.class);
+        ContentDisposition cd = mock(ContentDisposition.class);
+        when(cd.getParameter("filename")).thenReturn(filename);
+        when(attachment.getContentDisposition()).thenReturn(cd);
+        when(attachment.getObject(InputStream.class)).thenReturn(is);
+
+        List<Attachment> attachments = List.of(attachment);
+        Response uploadResp = eventConfRestApi.uploadEventConfFiles(attachments, securityContext);
+        assertEquals(Response.Status.OK.getStatusCode(), uploadResp.getStatus());
+
+        EventConfSource eventConfSource = eventConfSourceDao.findByName("Cisco.airespace");
+        Response response = eventConfRestApi.downloadEventConfXmlBySourceId(eventConfSource.getId(), securityContext);
+
+        assertNotNull("Response should not be null", response);
+        assertEquals("Expected HTTP 200 OK", 200, response.getStatus());
+
+        Object entity = response.getEntity();
+        assertNotNull("Response entity should not be null", entity);
+
+        String downloadedXml;
+        if (entity instanceof StreamingOutput) {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ((StreamingOutput) entity).write(baos);
+            downloadedXml = baos.toString(StandardCharsets.UTF_8);
+        } else if (entity instanceof InputStream) {
+            downloadedXml = new String(((InputStream) entity).readAllBytes(), StandardCharsets.UTF_8);
+        } else if (entity instanceof String) {
+            downloadedXml = (String) entity;
+        } else {
+            fail("Unexpected entity type: " + entity.getClass());
+            return;
+        }
+
+        assertFalse("Downloaded XML should not be empty", downloadedXml.isEmpty());
+        assertTrue("XML should end with </events>", downloadedXml.trim().endsWith("</events>"));
+
+        is = getClass().getResourceAsStream(path);
+        String uploadedXml = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+
+        Events uploaded = JaxbUtils.unmarshal(Events.class, new StringReader(uploadedXml));
+        Events downloaded = JaxbUtils.unmarshal(Events.class, new StringReader(downloadedXml));
+
+        final var uploadedEvents = new ArrayList<>(uploaded.getEvents());
+        final var downloadedEvents = new ArrayList<>(downloaded.getEvents());
+
+        Comparator<Event> eventComparator = Comparator
+                .comparing(Event::getUei, Comparator.nullsLast(String::compareTo))
+                .thenComparing(Event::getEventLabel, Comparator.nullsLast(String::compareTo))
+                .thenComparing(Event::getDescr, Comparator.nullsLast(String::compareTo));
+
+        uploadedEvents.sort(eventComparator);
+        downloadedEvents.sort(eventComparator);
+
+        assertEquals(uploadedEvents.size(), downloadedEvents.size());
```

Comment:
You can easily compare Events without bothering about order by doing simple  .equals on Event object

 for (Event uploadedEvent : uploadedEvents) {
      assertTrue(downloadedEvents.contains(uploadedEvent));
  }